### PR TITLE
Hw3

### PR DESCRIPTION
**Pull request by timand32**
1. The submissions in 4 modules this time: `app`, `data`, `calc`, and `transformers`, everything's in ./assignments/hw3/ this time.
2.  `app` is the main executable file.
3.  `data` loads and processes tables from the baseball dataset. Note that you may have to change some constants at the top if things aren't working.
4. `calc` performs the rolling average calculation.
5. `transformers.RollingBATransformer` is a pyspark.ml.Transformer that simply wraps `calc.calculate_ba()`.

### CODE REVIEW

File: assignments/hw3/app.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+"""
+import sys
+
+import data
+from pyspark.sql import SparkSession
+from transformers import RollingBATransform
+
+
+def main() -> int:
+    # Builda a Spark session
```
Comment (amypetris) [inline assignments/hw3/app.py]:
```suggestion
    # Builds a Spark session
```

File: assignments/hw3/calc.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+"""Module calculates rolling batting averages
+    from a 'master' list of stats per game.
+"""
+
+import sys
+
+from pyspark import StorageLevel
+from pyspark.sql import DataFrame, SparkSession
+
+ROLLING_SQL = """
```
Comment (amypetris) [inline assignments/hw3/calc.py:8]: A comment explaining this query may be helpful:+1:

File: assignments/hw3/calc.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    # since we have one table now, try out Spark SQL.
+    # https://spark.apache.org/docs/2.2.0/sql-programming-guide.html
+    mstr_df.createOrReplaceTempView("master")
+    mstr_df.persist(StorageLevel.DISK_ONLY)
+    rlng_df = spark.sql(ROLLING_SQL)
+    return rlng_df
+
+
+def main() -> int:
+    return 0
```
Comment (amypetris) [inline assignments/hw3/calc.py]: Is the main function needed in this file?

**Review by amypetris — COMMENTED** [review]
Everything looks good to me:+1: I was able to get everything running on my machine. I made a couple minor suggestions about documentation.

File: assignments/hw3/app.py (hunk 1 above)
Comment (timand32) [reply assignments/hw3/app.py]: Typo fixed in [`e172a9a`](https://github.com/timand32/bda696/pull/3/commits/e172a8a340ccb8ba32186cbd7a2b9fd5e2557ef7).

File: assignments/hw3/calc.py (hunk 2 above)
Comment (timand32) [reply assignments/hw3/calc.py:8]: Comment added in [`e172a9a`](https://github.com/timand32/bda696/pull/3/commits/e172a8a340ccb8ba32186cbd7a2b9fd5e2557ef7) (Thank you for catching this, I forgot to mention the code was adapted from the professor's query as well as giving a general explaination).

File: assignments/hw3/calc.py (hunk 3 above)
Comment (timand32) [reply assignments/hw3/calc.py]: It is not, removed it in [`e172a9a`](https://github.com/timand32/bda696/pull/3/commits/e172a8a340ccb8ba32186cbd7a2b9fd5e2557ef7).

**Pull request merged**